include/zcbor_diag_print.h

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZCBOR_DIAG_PRINT_H__
+#define ZCBOR_DIAG_PRINT_H__
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef ZCBOR_DIAG_PRINT_FUNC
+#include <zephyr/sys/printk.h>
+#define zcbor_do_diag_print(...) printk(__VA_ARGS__)
+#else
+#define zcbor_do_diag_print(...) ZCBOR_DIAG_PRINT_FUNC(__VA_ARGS__)
+#endif
+
+static bool indent_printed = false;
+
+__attribute__((used))
+static void zcbor_print_indent(size_t indent_len)
+{
+	if (!indent_printed) {
+		for (int i = 0; i < indent_len; i++) {
+			zcbor_do_diag_print("| ");
+		}
+		indent_printed = true;
+	}
+}
+
+__attribute__((used))
+static void zcbor_print_newline(void)
+{
+	zcbor_do_diag_print("\r\n");
+	indent_printed = false;
+}
+
+#define BYTES_PER_LINE 16
+
+__attribute__((used))
+static void zcbor_print_str(const uint8_t *str, size_t len, size_t indent_len)
+{
+	for (size_t i = 0; i < len; i++) {
+		if (!(i % BYTES_PER_LINE)) {
+			if (i > 0) {zcbor_print_newline();}
+			zcbor_print_indent(indent_len);
+			zcbor_do_diag_print("0x");
+		}
+		zcbor_do_diag_print("%02x ", str[i]);
+	}
+}
+
+__attribute__((used))
+static void zcbor_print_bstr(uint8_t const *payload, size_t len, size_t indent_len)
+{
+	zcbor_print_str(payload, len, indent_len);
+	zcbor_print_newline();
+}
+
+static void zcbor_print_tstr(uint8_t const *payload, size_t len, size_t indent_len);
```

Comment:
THe file is still called `zcbor_diag_print.h` should we change it? IMO it could stay the way it is.

Reply:
Changed